agents/builtin/mongodb/internal/profiler/profiler_test.go

```diff
@@ -15,3 +15,74 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 package profiler
+
+import (
+	"testing"
+	"time"
+
+	"github.com/percona/pmgo"
+	"github.com/percona/pmm/api/qanpb"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/mgo.v2/bson"
+
+	"github.com/percona/pmm-agent/agents/builtin/mongodb/internal/profiler/aggregator"
+	"github.com/percona/pmm-agent/agents/builtin/mongodb/internal/report"
+)
+
+func TestProfiler(t *testing.T) {
+	defer func() {
+		aggregator.DefaultInterval = time.Duration(time.Minute)
+	}()
+	aggregator.DefaultInterval = time.Duration(time.Second)
+	url := "mongodb://pmm-agent:root-password@127.0.0.1:27017"
+
+	dialInfo, err := pmgo.ParseURL(url)
+	require.NoError(t, err)
+
+	dialer := pmgo.NewDialer()
+
+	sess, err := createSession(dialInfo, dialer)
+	require.NoError(t, err)
+
+	err = sess.DB("test").DropDatabase()
+	require.NoError(t, err)
+
+	ms := &testWriter{t: t}
+	prof := New(dialInfo, dialer, logrus.WithField("component", "profiler-test"), ms, "test-id")
+	err = prof.Start()
+	require.NoError(t, err)
+
+	err = sess.DB("test").C("peoples").Insert(bson.M{"name": "Anton"}, bson.M{"name": "Alexey"})
+	require.NoError(t, err)
```

Comment:
That one better be be `assert`, otherwise we call `prof.Start()`, but do not call `prof.Stop()` if there is insertion error.